libraries/Microsoft.Bot.StreamingExtensions/RequestEx.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+
+namespace Microsoft.Bot.StreamingExtensions
+{
+    /// <summary>
+    /// Implementation split between Request and RequestEx.
+    /// The basic request type sent over Bot Framework Protocol 3 with Streaming Extensions transports,
+    /// equivalent to HTTP request messages.
+    /// </summary>
+    public partial class Request
+    {
```

Comment:
Calling a class "Request" is going to cause many problems. Please come up with a better name.

StreamingRequest. BotFrameworkRequest. BotFrameworkProtocolRequest.

Reply:
But it is a request. We can consider adding a prefix to it, but that's kind of the point of namespaces, no?